server/test/sc/server/plugins/TestGame.kt

```diff
@@ -0,0 +1,103 @@
+package sc.server.plugins
+
+import org.slf4j.LoggerFactory
+import sc.api.plugins.IGameState
+import sc.api.plugins.exceptions.TooManyPlayersException
+import sc.framework.plugins.ActionTimeout
+import sc.framework.plugins.Player
+import sc.framework.plugins.RoundBasedGameInstance
+import sc.protocol.responses.ProtocolMessage
+import sc.server.helpers.TestTeam
+import sc.server.helpers.WinReason
+import sc.shared.PlayerScore
+import sc.shared.ScoreCause
+import sc.shared.WelcomeMessage
+import sc.shared.WinCondition
+import java.util.*
+
+class TestGame : RoundBasedGameInstance<TestPlayer?>(TestPlugin.TEST_PLUGIN_UUID) {
+    private val state = TestGameState()
+    override fun onRoundBasedAction(fromPlayer: Player, data: ProtocolMessage) {
+        /*
+         * NOTE: Checking if right player sent move was already done by
+         * {@link sc.framework.plugins.RoundBasedGameInstance#onAction(Player, Object)}.
+         * There is no need to do it here again.
+         */
+
+        if (data is TestMove) {
+            data.perform(state)
+            next(if (state.currentPlayer === TestTeam.RED) state.red else state.blue)
+        }
+    }
+
+    override fun checkWinCondition(): WinCondition? {
+        return if (this.round > 1) {
+            logger.info("Someone won")
+            WinCondition(if (state.state % 2 == 0) TestTeam.RED else TestTeam.BLUE, WinReason.ROUND_LIMIT_FREE_FIELDS)
+        } else null
+    }
+
+    @Throws(TooManyPlayersException::class)
```

Comment:
you can usually remove that in Kotlin

Reply:
probably, yes, but I wouldn't know. I'm not aware of what java code calls what kotlin code.
Just because this isn't necessary doesn't mean it makes the code unclean

Reply:
removing the @Throws will never inhibit compilation, so unless the exception is particularly important to be highlighted (which in this case it isn't, since the case of too many players joining is atypical), just remove it

Reply:
included in 2c7b1675